File: python/lights.py
#!/usr/bin/env python

import math
import unittest

from color import Color
from tuples import Tuple, Point, Vector

class PointLight:
    def __init__(self, position, intensity):
        self.position = position
        self.intensity = intensity

class TestPointLight(unittest.TestCase):
    def test_point_light_constructor(self):
        intensity = Color(1,1,1)
        position = Point(0,0,0)
        light = PointLight(position, intensity)

        self.assertEqual(light.position, position)
        self.assertEqual(light.intensity, intensity)

File: src/lights.c
#include "rtc.h"

void light_init(light *o, const v4 position, const v3 intensity)
{
  memcpy(o->position, position, sizeof(v4));
  memcpy(o->intensity, intensity, sizeof(v3));
}

void point_light_init(light *o, const v4 position, const v3 intensity)
{
  o->type = PointLightType;
  light_init(o, position, intensity);
}

